Added AI usage guideline for the application process (#118)
HackYourFuture-CPH/programme: guidelines/ai-usage.md

# Using Generative AI at HackYourFuture

> During your time at HackYourFuture, whether you're applying to join and in the application process, or already on the programme, you may wish to use Generative AI (AI) in some form to help you out on a task. Using AI includes using tools such as ChatGPT, GitHub Copilot and similar. This guideline aims to explain how HackYourFuture views AI usage, and our expectations for how you should (and should not) use it while working with us.

Reviewer: Sounded a bit strange to me, and changed the order of the first sentence.
```suggestion
Whether you're applying to join HackYourFuture or are already enrolled in the programme, you may consider using Generative AI tools to solve various problems. This includes tools like ChatGPT, GitHub Copilot, and similar. This guideline outlines how HackYourFuture views the use of AI and our expectations for how you should (and should not) use it while working with us.
```

> [!WARNING]

> > It’s important you follow these guidelines. If we find evidence of you misusing AI based on the below, you may be rejected from the application process or removed from the programme.

Reviewer: ```suggestion
> It’s important that you follow these guidelines. If we find evidence that you’ve misused AI as outlined below, you may be rejected from the application process or removed from the programme."
```

## How you can use AI

> Used effectively, AI can be a great tool to support your learning process and skill development. However, if you're not careful, it can hinder your progress and take away important opportunities to learn.

Reviewer: We've used "journey" a few times already across the programme
```suggestion
Used effectively, AI can be a great tool to support your learning journey and skill development. However, if you're not careful, it can hinder your progress and take away important learning opportunities.
```

   e.g. "I'm printing an error using 'console.log', are there alternatives I could use?"
1. Getting feedback on your communication, messages and documentation
   e.g. "I'm writing a message to my teammate to ask for a favour, is there a more thoughtful way to phrase my last sentence?"

### Bad use cases (these are not allowed)

1. Generating full coding solutions or functions
   e.g. "I need to build a todo app for my assignment, please generate the javascript code for this"
1. Generating code comments or explanations of the solutions you create (they should be written by you, based on your own understanding)
   e.g. "Add code comments to my code to explain what every function does"
1. Generating your written communication for you (e.g. application form, cover letters, slack messages)
   e.g. "I'm applying to HackYourFuture, please write a cover letter for me that I can send with my application"
1. During the interview to help you answer questions (unless the interviewer asks you to use AI)
   e.g. Secretly typing the questions we ask you into ChatGPT, and sharing those answers with us as your own

## How your experience and knowledge will be evaluated

> While you may choose to use AI (following the guidelines above), please be aware that **you will be required to be able to explain all of your answers and code** during your time at HackYourFuture. Make sure when you create, submit or share code or solutions, that you fully understand how it works based on your own knowledge.

Reviewer: Softened the language a bit by changing "required" to "expected".
Moved "make sure" to more appropriate place
```suggestion
While you're welcome to use AI tools (following the guidelines above), please be aware that **you'll be expected to clearly explain all your answers and code** during your time at HackYourFuture. Whenever you create, submit, or share code or solutions, make sure you fully understand how they work based on your own knowledge.
```
